src/main/java/org/mskcc/cbio/oncokb/web/rest/AccountResource.java

```diff
@@ -274,6 +276,29 @@ public void changePassword(@RequestBody PasswordChangeDTO passwordChangeDto) {
         userService.changePassword(passwordChangeDto.getCurrentPassword(), passwordChangeDto.getNewPassword());
     }
 
+    /**
+     * {@code POST  /account/request-api-access} : requests ROLE_API for current user.
+     *
+     * @param apiAccessRequest isRequested boolean and justification (reason for request).
+     */
+    @PostMapping(path = "/account/request-api-access")
+    public void requestApiAccess(@RequestBody ApiAccessRequest apiAccessRequest) {
+        Optional<String> userLogin = SecurityUtils.getCurrentUserLogin();
```

Comment:
We should probably add a check to see if the justification is not empty here using an if condition. You can also add `@NotEmpty` annotation to the ApiAcessRequest's `justification` property.

Reply:
Set @NotEmpty to the justification property so it should never be empty. I think that's enough given an API request cannot come without a justification?

Reply:
@bprize15 does @NonEmpty through exception when it's actually empty? I remember the annotation here mostly for displaying on swagger. It actually doesn't enforce the content to be nonempty.

Reply:
@zhx828 I added `@Valid` to the definition of requestApiAccess method like so:

`@PostMapping(path = "/account/request-api-access") public void requestApiAccess(@Valid @RequestBody ApiAccessRequest apiAccessRequest)`

If the justification is empty, the controller will return 400

Reply:
I was thinking that the user could technically use something like postman to send a request with justification set to empty string.

Would like to know if the annotation throws error too.

Reply:
I tested and if they use Postman they will get a 400 response. I also allowed users to send empty justifications temporarily through the website, and the post request was throwing an error